Sources/PackageCollections/Providers/JSONPackageCollectionProvider.swift

```diff
@@ -0,0 +1,364 @@
+        return options
+    }
+
+    public struct Configuration {
+        public var maximumSizeInBytes: Int
+        public var requestTimeout: DispatchTimeInterval?
+
+        public init(maximumSizeInBytes: Int? = nil,
+                    requestTimeout: DispatchTimeInterval? = nil) {
+            // TODO: where should we read defaults from?
+            self.maximumSizeInBytes = maximumSizeInBytes ?? 5_000_000 // 5MB
+            self.requestTimeout = requestTimeout ?? .seconds(1)
+        }
+    }
+
+    public enum Errors: Error {
+        case invalidJSON(Error)
+        case invalidResponse(String)
+        case responseTooLarge(Int)
+    }
+}
+
+// MARK: - Package feed models
+
+extension JSONPackageCollectionProvider {
+    fileprivate struct CollectionV1: Decodable {
+        let name: String
+        let description: String?
+        let keywords: [String]?
+        let formatVersion: String
+        let revision: Int?
+        let packages: [Package]
+        let generatedAt: Date
+        let generatedBy: Author?
+
+        private enum CodingKeys: String, CodingKey {
+            case name
+            case description
+            case keywords
+            case formatVersion
+            case revision
+            case packages
+            case generatedAt
+            case generatedBy
+        }
+    }
+}
+
+extension JSONPackageCollectionProvider.CollectionV1 {
+    fileprivate struct Package: Decodable {
+        let url: String
+        let description: String?
+        let keywords: [String]?
+        let versions: [Version]
+        let readmeURL: String?
+
+        enum CodingKeys: String, CodingKey {
+            case url
+            case description
+            case keywords
+            case versions
+            case readmeURL
+        }
+    }
+
+    fileprivate struct Version: Decodable {
+        let version: String
+        let packageName: String
+        let toolsVersion: String
+        let targets: [Target]
+        let products: [Product]
+        let minimumPlatformVersions: [PlatformVersion]?
+        let verifiedPlatforms: [Platform]?
+        let verifiedSwiftVersions: [String]?
+        let license: License?
+
+        enum CodingKeys: String, CodingKey {
+            case version
+            case packageName
+            case toolsVersion
+            case targets
+            case products
+            case minimumPlatformVersions
+            case verifiedPlatforms
+            case verifiedSwiftVersions
+            case license
+        }
+    }
+
+    fileprivate struct Target: Decodable {
+        let name: String
+        let moduleName: String?
+    }
+
+    fileprivate struct Product: Decodable {
+        let name: String
+        let type: ProductType
+        let targets: [String]
+
+        enum CodingKeys: String, CodingKey {
+            case name
+            case type
+            case targets
+        }
+    }
+
+    fileprivate enum ProductType: Decodable {
```

Comment:
We don't need this since we should be using `PackageModel.ProductType`, which already conforms to `Codable`.